### PR TITLE
Customer reviews

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

**Comment by andywerner** [comment]
I know I said you should use the laser safety screen as a template… But now I found that there is an easier way to read user settings on the frontend from the LoginStateViewModel: (here accessed via workinAreaViewModel:)
`mrbeam.viewModels.workingAreaViewModel.loginState.currentUser()`

I’d like to discuss with you tomorrow if it makes sense to refactor your code for the reviews so that it takes advantage of this.
One potential pitfall of this above is that I don’t know when/if it updates during runtime.

**Comment by andywerner** [comment]
Theres also the userSettingsViewModel:
https://docs.octoprint.org/en/master/plugins/viewmodels.html#dependencies

**Comment by andywerner** [comment]
`mrbeam.viewModels.userSettingsViewModel.updateSettings(mrbeam.viewModels.workingAreaViewModel.loginState.currentUser().name, {mrbeam:{andytest:'hallo'}})`

this one wipes all mrbeam suer settings. So a logged in suer can add any settings he wants.

File: octoprint_mrbeam/__init__.py [hunk 1]
```diff
@@ -292,6 +295,7 @@ def get_settings_defaults(self):
 			),
 			grbl_version_lastknown=None,
 			tour_auto_launch = False,
+			num_succ_jobs = 0,
```
Comment (andywerner) [inline octoprint_mrbeam/__init__.py]: move to usage,yaml

File: octoprint_mrbeam/__init__.py [hunk 2]
```diff
@@ -1249,6 +1259,16 @@ def on_api_command(self, command, data):
...
 
+	def get_user_name(self):
+		from flask.ext.login import current_user
+
+		# Looks like current_user sometimes does not work, so we save it and the next time if there's no information
+		# we just use the last saved user.
+		if current_user and not current_user.is_anonymous():
+			self._current_user = current_user.get_name()
+
+		return self._current_user
```
Comment (andywerner) [inline octoprint_mrbeam/__init__.py:1277]: not needed any more

File: octoprint_mrbeam/static/js/review.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        self.justGaveReview = ko.observable(false);
+
+        self.shouldAskForReview = ko.computed(function(){
+            if (self.loginState.currentUser() && self.loginState.currentUser().active) {
+                let numSuccJobs = self.loginState.currentUser().settings.mrbeam.review.num_succ_jobs;
+                let reviewGiven = self.loginState.currentUser().settings.mrbeam.review.given;
+
+                console.log('############ NUM JOBS: '+numSuccJobs);
+                console.log('############ GIVEN: '+reviewGiven);
+                console.log('############ NOW YES: '+self.justGaveReview());
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/review.js]: remove

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/__init__.py (hunk 2 above)
Comment (irlaec) [reply octoprint_mrbeam/__init__.py:1277]: I actually need it...

File: octoprint_mrbeam/__init__.py (hunk 1 above)
Comment (irlaec) [reply octoprint_mrbeam/__init__.py]: moved

File: octoprint_mrbeam/static/js/review.js (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/review.js]: removed

**Pull request merged**